src/lib.rs

```diff
@@ -285,6 +289,78 @@ impl Into<Affine> for PublicKey {
     }
 }
 
+impl Serialize for PublicKey {
+    fn serialize<S>(&self, serializer: S) -> Result<S::Ok, S::Error>
+        where S: Serializer
+    {
+        serializer.serialize_str(&base64::encode(&self.serialize()[..]))
```

Comment:
I'm just wondering whether this is general enough -- here we are making the assumption that public keys are always serialized to string, which is something that might not hold for things like binary formats.

@fubuloubu How about creating a wrapper type for PublicKey and implement `serde` to `str` there?

Reply:
I guess if a `PublicKey`'s bytes methods are available, this would work fine.

Basically, between this and the above comment, should I just removed serialization from these commits?

Reply:
I tink the solution here is to use [`is_human_readable()`](https://docs.serde.rs/serde/trait.Serializer.html#method.is_human_readable). 

For example see https://docs.rs/secp256k1/0.17.2/src/secp256k1/lib.rs.html#421-446